### PR TITLE
feat(FormControls): updated logic for rendering icons

### LINKED ISSUE #9023: Form control - restructure to support removal of background images icons
**Is this a new component or an extension of an existing one?**
This applies to form controls that are using the native elements for text input, select, and text area.

**Describe the feature**
For https://github.com/patternfly/patternfly/issues/5439 the method of using a background image on the control to display an icon needs to be changed. Per leads meeting of 4/11/23, a decision was made to remove the background image, and instead add a wrapper element around the native elements with another child element able to hold multiple icons. The wrapper element will receive the PatternFly styling and the HTML elements themselves will have appearance:none to remove browser-specific styling.

More details to come when the PR for the core issue https://github.com/patternfly/patternfly/issues/5469 is in place.

### PR DESCRIPTION
**Pull request by thatblindgeye**
**What**: Closes #9023 

@srambach @mcarrano currently the Core [examples for Form Control](https://patternfly-pr-5531.surge.sh/components/forms/form-control#input) shows a Textinput with both a custom icon and a status icon. I put this functionality in React's TextInput, but notTextArea or FormSelect. Do we want to allow multiple icons to be rendered, and if so should it be on all 3 of these components?

Other notes:
- deprecated Select and TreeViewSearch were both using inputs with the `pf-v5-c-form-control` class. These both (or just TreeView?) will end up needing to be updated to use SearchInput instead. For now the styling for TreeViewSearch is different as I hardcoded in markup to render the search icon on the right (rather than the left).
- The autoResize logic in TextArea currently does not work
- Added a FromControlIcon helper component rather than adding a `___Icon` sub-component for TextArea,FormSelect, and TextInput. I can move this to a more applicable directory if need be.

**Additional issues**:

### CODE REVIEW

**Comment by nicolethoen** [comment]
And this will need a code mod, right? 

- props being removed related to iconSprites
- props being changed related to customIcons
- the TextInput's isReadOnly prop being removed
- the new way to handle expandable text areas
- probably more I missed?

**Comment by nicolethoen** [comment]
> * The autoResize logic in TextArea currently does not work

@thatblindgeye is this a regression? Is it because of the new DOM structure? Can we fix it w/o a breaking change and maybe make a follow up issue or does it require investigation today?

**Comment by thatblindgeye** [comment]
@nicolethoen Yeah there'll definitely be some codemods needed after this update. I'm going to go through and open an issue(s) for anything I catch.

And I believe it *might* be due to new structure, but I'm going to investigate a bit more into it.

**Comment by srambach** [comment]
> Do we want to allow multiple icons to be rendered, and if so should it be on all 3 of these components?
I don't know of any place that uses an icon (not a status icon) on a text area or select. @mcoker?
The text input definitely needs to support both a custom icon and a status.

> The autoResize logic in TextArea currently does not work
This is almost definitely due to the new structure - in order to make sure the icon doesn't float off by itself, the resize was moved out onto the wrapper div rather than the text area itself. Then the text area just takes up as much room as there is. Is there a way to manage auto resize with that?

**Comment by thatblindgeye** [comment]
@nicolethoen @srambach Updated the autoresizing logic. Looks like the previous `heightToken` isn't actually applied to anything in Core anymore, so now just setting the `height` on the `.pf-v5-c-form-control` div.

**Comment by srambach** [comment]
File upload changed a little

**Comment by mcoker** [comment]
@mcarrano @srambach @thatblindgeye Looks like we're styling the old form control class for things that need to be updated with to style the right element based on the updates made in core to remove the background images - https://github.com/patternfly/patternfly/issues/5583

**Comment by thatblindgeye** [comment]
@mcarrano I updated the icon on TimePicker. Looks like the other issues should be covered by the issue @mcoker created in Core above.

**Review by nicolethoen — APPROVED** [review]
the react code here looks solid

**Review by mcarrano — APPROVED** [review]
> @mcarrano I updated the icon on TimePicker. Looks like the other issues should be covered by the issue @mcoker created in Core above.

In that case I approve.

**Review by wise-king-sullyman — APPROVED** [review]

File: packages/react-core/src/helpers/FormControlIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  /** A custom icon to render instead of a status icon. */
+  customIcon?: React.ReactNode;
+  /** The status icon to render. */
+  status?: 'success' | 'error' | 'warning';
+}
+
+export const FormControlIcon = ({ status, customIcon, className, ...props }: FormControlIconProps) => {
+  const StatusIcon = status && statusIcons[status];
+
+  return customIcon || StatusIcon ? (
```
Comment (tlabaj) [inline packages/react-core/src/helpers/FormControlIcon.tsx:26]: I do not know that this check is necessary this is an internal component.

**Review by tlabaj — CHANGES_REQUESTED** [review]

**Review by gitdallas — APPROVED** [review]

**Comment by tlabaj** [comment]
> @mcarrano @srambach @thatblindgeye Looks like we're styling the old form control class for things that need to be updated with to style the right element based on the updates made in core to remove the background images. That will account for some of these visual bugs you're seeing @mcarrano. I created an issue to fix those - [patternfly/patternfly#5583](https://github.com/patternfly/patternfly/issues/5583)

@mcoker is this good to merge as is or do do we want to wait for core bug to be fixed?

**Comment by mcoker** [comment]
@tlabaj no need to wait for core IMO, those updates will not need a react follow up.

**Review by kmcfaul — APPROVED** [review]

**Review by mcoker — APPROVED** [review]
LGTM! Just one note about a modifier on `<TextInput>` that I don't think is necessary.

**Review by tlabaj — APPROVED** [review]
LGTM!

**Pull request merged**